manifests/install.pp

```diff
@@ -142,9 +169,39 @@
           }
         }
       }
-      package { $python_virtualenv: ensure => $venv_ensure }
-      package { $pip: ensure => $pip_ensure }
-      package { $pythondev: ensure => $dev_ensure }
+
+      if $::osfamily == 'FreeBSD' {
+        if $pip_ensure == present and $::python::version =~ /^3/ {
+          # https://docs.python.org/3.4/library/ensurepip.html
+          exec { 'install_pip34':
+            command => '/usr/local/bin/python3.4 -m ensurepip',
+            creates => '/usr/local/bin/pip3.4',
+            require => Package[$python],
+          }
+
+          file { '/usr/local/bin/pip':
+            ensure => link,
+            target => '/usr/local/bin/pip3.4'
+          }
+        }
+      }
+
+      if $python_virtualenv {
+        if $::osfamily == 'FreeBSD' and $::python::version =~ /^3/ {
+          package { $python_virtualenv:
+            ensure   => $venv_ensure,
+            provider => 'pip',
+          }
+        } else {
+          package { $python_virtualenv: ensure => $venv_ensure }
+        }
```

Comment:
not a big fan of this if-pyramid.
can't we just add a variable `$provider` and override it?
that would make for more declarative looking code.